### ISSUE TITLE
Install swig from package manager

### ISSUE THREAD

**Issue by Aposhian** [Fields2Cover/Fields2Cover#143]
The docker build that builds the python interface builds swig from source. Is there a reason for that? It seems to work just fine when installing swig as a binary package (ex: via apt).

For ROS users using `rosdep`, adding `swig` to the `package.xml` would then automate this dependency installation (and also this would apply for bloom).

**Comment by Aposhian** [comment]
Thanks for the explanation on why swig is being installed from source. I'm using Ubuntu 22.04 which has GDAL 3.4 and Swig 4.0 as the latest packaged versions and those seem to be working fine.

If we add `swig` to the package.xml, it doesn't require the python interface to be compiled. It will pull it in as a bloom dependency and it will install it for users with rosdep, but doesn't require anything to be compiled.

I would think that it makes sense for the ROS bloomed package to have the python interface built anyway, and if users don't want to build it then they can build from source.

Another alternative would be splitting the python interface into a separate package internally, and that would contain the swig dependency.

**Comment by tonynajjar** [comment]
I created https://github.com/Fields2Cover/Fields2Cover/pull/144 before seeing this issue.

> I would think that it makes sense for the ROS bloomed package to have the python interface built anyway

From a user perspective, I'd love that as well so I don't have to build it myself
The next best thing is adding swig in package.xml so that I don't have to do it manually outside of the rosdep workflow

**Comment by Gonzalo-Mier** [comment]
#144 merged! Should we turn on the python compilation flag by default?

**Comment by Aposhian** [comment]
I think with that PR merged we call this complete

**Comment by Gonzalo-Mier** [comment]
Done :smile:

**Comment by tonynajjar** [comment]
Hey guys, did any of you confirm that this work? 

`docker run -it --rm osrf/ros:humble-desktop-full bash -ic 'sudo apt update && sudo apt install ros-humble-fields2cover -y && python3 -c "import field2cover"'`

This should theoretically work no? But it doesn't -> `ModuleNotFoundError: No module named 'field2cover'`

**Comment by vinnnyr** [comment]
@tonynajjar you are right, I checked on Humble a few different ways, and it seems as if though only the cpp library is installed via apt.

**Comment by tonynajjar** [comment]
Thank you for checking, let's at least reopen this ticket to keep track of it
